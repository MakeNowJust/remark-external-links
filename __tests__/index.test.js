/* eslint-env jest */

var remark = require('remark')
var html = require('remark-html')
var externalLinks = require('..')

var input = '[remark](https://github.com/remarkjs/remark)'

test('should add the defaults when without options', async () => {
  expect(
    remark()
      .use(externalLinks)
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should do nothing if both are set to false', async () => {
  expect(
    remark()
      .use(externalLinks, {target: false, rel: false})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should add a target', async () => {
  expect(
    remark()
      .use(externalLinks, {target: '_blank', rel: false})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should add a rel', async () => {
  expect(
    remark()
      .use(externalLinks, {target: false, rel: 'nofollow'})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should add both', async () => {
  expect(
    remark()
      .use(externalLinks, {target: '_blank', rel: 'nofollow'})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})
