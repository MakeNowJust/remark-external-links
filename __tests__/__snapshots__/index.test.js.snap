// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a rel 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" rel=\\"nofollow\\">remark</a></p>
"
`;

exports[`should add a target 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\">remark</a></p>
"
`;

exports[`should add both 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\" rel=\\"nofollow\\">remark</a></p>
"
`;

exports[`should add the defaults when without options 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\" rel=\\"nofollow noopener noreferrer\\">remark</a></p>
"
`;

exports[`should do nothing if both are set to false 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\">remark</a></p>
"
`;
